Yoast/wordpress-seo
# Add get related keyphrase endpoint

## Context


* This PR introduces a new endpoint that sends a request to SEMrush with the given keyphrase and database (country code) so we can properly retrieve information for the SEMrush modal.

## Summary


This PR can be summarized in the following changelog entry:

* Introduces an API endpoint to retrieve data related to the keyphrase, from the SEMrush API.

## Relevant technical choices:

* This PR introduces some refactoring of code that somehow got broken along the way (the `token` prop in the SEMrush client). It also introduces a different approach to performing GET requests, as the OAuth library doesn't have native support for sending along query parameters.

## Test instructions

This PR can be tested by following these steps:

* Ensure you have a valid SEMrush token present in your database before testing (see https://github.com/Yoast/wordpress-seo/pull/15607 for more information on how to do this)
* In Postman (or a similar application), add a new GET request to http://basic.wordpress.test/wp-json/yoast/v1/semrush/related_keyphrases
* In the Params tab of Postman, add the following two params: `keyphrase` and `database`. You're free to choose whatever value for the keyphrase. I suggest using something like `nl` or `us` for the database.
* In the Authorization tab, ensure you've selected Basic Auth and fill in the admin credentials for your development environment.
* Hit Send. You should get a response back from SEMrush containing various data on related keyphrases.
* Check your database and see that in `wp_options` a new entry was created named `_transient_wpseo_semrush_related_keyphrases_<your keyphrase>_<your selected database>`.
* Make sure to also test with no (or faulty) admin credentials to ensure the permission callback successfully returns an error message.

## UI changes

## Documentation

## Quality assurance
